apps/api-extractor/src/collector/Collector.ts

```diff
@@ -826,10 +829,14 @@ export class Collector {
               astSymbol
             );
           }
+
+          // All internal, exported APIs default to public if no release tag is specified.
+          options.effectiveReleaseTag = ReleaseTag.Public;
         }
+      } else {
+        // All external APIs default to public if no release tag is specified.
+        options.effectiveReleaseTag = ReleaseTag.Public;
```

Comment:
This change is to prevent unexported items included in an API report from being assigned the `@public` release tag if none is specified.

Reply:
After some discussion, I think this is not right:  The effective release tag must always be something other than `ReleaseTag.None`.  

The simplest approach would be to handle forgotten exports like any other API item:
- API Extractor complains if it doesn't have a release tag (because for example .d.ts trimming is not going to produce good results if developers don't think carefully about what is/isn't public)
- If you disable the warnings, then everything defaults to `@public` (because in this case, the developer presumably doesn't care very much about trimming or release tags)

Reply:
Updated this PR to match the behavior you described above. However, I was thinking a bit more about forgotten exports and release tags and sent you a message on Zulip, reproduced below:

> We decided yesterday that the missing release tag warning should only be logged if includeForgottenExports is on. I'm wondering now if the missing release tag warning should be logged regardless of whether that config is on. Why? Because .d.ts trimming will produce better results if forgotten exports have release tags.
>
> Maybe the original thinking was that a developer will already get an ae-forgotten-export warning for these items, and so there's already "some warning" indicating that these items should be exported (and thus also should have release tags). But (1) what if the developer turned off that warning and (2) even if the developer didn't, it still might be useful to report both warnings (otherwise a developer might slap on an export, re-run API Extractor, and then just hit another warning right after, as they forgot to also add a release tag).

Reply:
Agreed, but I think we could implement that extra validation in a separate PR.